src/source.js

```diff
@@ -0,0 +1,98 @@
+import * as util from "./util";
+
+function getJSON(url, callback) {
+  const xhr = new XMLHttpRequest();
+  xhr.responseType = "json";
+  xhr.open("get", url, true);
+  xhr.onload = function() {
+    if (xhr.status >= 200 && xhr.status < 300) {
+      callback(xhr.response);
+    } else {
+      throw new Error(xhr.statusText);
+    }
+  };
+  xhr.send();
+}
+
+export default url => {
+  /**
+   * A note on how this works:
+   * 0. At any moment we can recieve a request for a tile.
+   * 1. Before we can fulfil such a request, we need to load metadata. So we store tile requests that were issued before
+   *    metadata was loaded and once it loads we issue requests for the tiles once that is done.
+   * 2. If metadata is loaded, we check if there already has been a request for the same tile. If yes, we simply add
+   *    the callback to the queue, otherwise we save the callback and load the image.
+   * 3. When an image is loaded we store the data in a cache and empty the queue of all relevant callbacks by calling them.
+   * 4. If there is already data in the cache, simply call the callback right away.
+   */
+  let tileRequests = {};
+  let data;
+  let requestsBeforeMetadataLoaded = new Set();
+  let cache = {};
+
+  getJSON(url, windData => {
+    data = windData;
+    requestsBeforeMetadataLoaded.forEach(coords => {
+      if (cache[coords]) {
+        let req;
+        while ((req = tileRequests[coords].pop())) {
+          dispatchReq(coords, req);
+        }
+      } else {
+        load(...coords.split("/"));
+      }
+    });
+    requestsBeforeMetadataLoaded = [];
+  });
+
+  function dispatchReq(coords, req) {
```

Comment:
No point shortening words, doesn't help comprehension, `dispatchRequest` is much easier to follow.